### DISCUSSION TITLE
Why is the process not terminated after release() ?

### DISCUSSION THREAD

**Discussion by GreenFlag31** [sidorares/node-mysql2#2565] [General]
```
try {
  const pool = mysql
    .createPool({
      host: '127.0.0.1',
      user: 'root',
      password: 'root',
      port: '3306',
      database: 'db_example',
    })
    .promise();

  const result = await pool.query('SELECT * FROM node_table');
  const rows = result[0];
  console.log(rows);

  const connection = await pool.getConnection();
  connection.release();
  // process.exit();
} catch (err) {
  console.error(err);
}

```

After I'm running this code, I'm unable to type anything in the terminal, the process is not terminated even if I'm using release(). I'm forced to use process.exit().
![image](https://github.com/sidorares/node-mysql2/assets/84795623/3a857213-02b3-49ee-bcb0-3282a06767f9)

Is it normal? (I'm new to mysql2)

Thank you

**Comment by wellwelwel** [comment]
Hi, @GreenFlag31 🙋🏻‍♂️

It's because you've released the connection, but the `pool` is still open.

You can end your `pool` using `await pool.end()`.
